impress-org/givewp-next-gen
# New: Next Gen Stripe Gateway

## Description



To really put the Next Gen gateway api to the test, we need a tangible gateway.  Since Stripe is usually the most involved in terms of front-end, we chose to scaffold a new Stripe gateway using the [Payment Element api](https://stripe.com/docs/payments/payment-element).  The PR adds the new gateway and will provide enough functionality to complete a donation on the next gen form.  There is still more polish to do in regards to the full transaction cycle, including receipts - but those pieces will come soon after this.

**JS Gateway Updates**
- New method `afterCreatePayment?(response: object): Promise<void> | Error` that can be used after creating the donation & returning from `createPayment($donation)` from the gateway.



## Affects



- The js gateway api
- Replace axios with fetch

## Visuals



<img width="633" alt="Screen Shot 2022-06-03 at 2 51 17 PM" src="https://user-images.githubusercontent.com/10138447/171929176-be506ac2-342d-4758-80f9-65a9729341f6.png">


## Testing Instructions



- Run `npm run dev`
- Make sure your stripe account is setup & next gen stripe enabled
- Donate using the stripe gateway using the next gen donation form block

## Pre-review Checklist
